snapcraft/tests/test_sources.py

```diff
@@ -68,6 +71,89 @@ def test_pull_tarball_must_download_to_sourcedir(self, mock_prov):
         with open(os.path.join(dest_dir, tar_file_name), 'r') as tar_file:
             self.assertEqual('Test fake compressed file', tar_file.read())
 
+    def test_checksum(self):
+        tar = tarfile.open("checksum.tar", "w")
+        tar.close()
+
+        # md5
+        source_checksum = '1276481102f218c981e0324180bafd9f'
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # sha1
+        source_checksum = '34e163be8e43c5631d8b92e9c43ab0bf0fa62b9c'
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # sha224
+        source_checksum = ('82be34614e8ca3d20d1733e52ae9b5b12902c196eeb4ee36c'
+                           '2625b5f')
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # sha256
+        source_checksum = ('84ff92691f909a05b224e1c56abb4864f01b4f8e3c854e4bb'
+                           '4c7baf1d3f6d652')
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # sha384
+        source_checksum = ('a31859d12a1176d02c2552f1bea2cba40a3ec809a7b224425'
+                           '04e7dd1c19201e6eeffce9a54d13760924ad73aad45049f')
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # sha512
+        source_checksum = ('1e543b135acb1da2d9ce119c11d6fa9de2c9ca2e97e55fdf2'
+                           '481c2944779a3d6df4a7c74f87692072ada4d494bbc3018d7'
+                           '545b3c631dac1bfb787f81e0b76530')
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # From file
+        source_checksum_write = open('CHECKSUM', 'w')
+        source_checksum_write.write('1e543b135acb1da2d9ce119c11d6fa9de2c9ca2e'
+                                    '97e55fdf2481c2944779a3d6df4a7c74f8769207'
+                                    '2ada4d494bbc3018d7545b3c631dac1bfb787f81'
+                                    'e0b76530 checksum.tar')
+        source_checksum_write.close()
+        source_checksum = 'CHECKSUM'
+
+        sources.check_checksum_determine_format(
+            source_checksum, "checksum.tar")
+
+        # From remote file
+        source_checksum = 'http://tsimonq2.net/misc/snapcraft-checksum-tar'
```

Comment:
@sergiusens I'm a little confused as to the usage of this, where can I find an example of a test case that uses it?